### DISCUSSION TITLE
Next.js and Headless CMS

### DISCUSSION THREAD

**Discussion by shadcn** [vercel/next.js#33087] [Ideas]
This is a curated list of issues, discussions and PRs that are related to improving the Next.js DX when working with headless CMS.

**If you find anything relevant, feel free to add it to the comments below.**

## Additional data from getStaticPaths to getStaticProps.

- https://github.com/vercel/next.js/discussions/11272
- https://github.com/vercel/next.js/discussions/33027
- @ijjk comment about caching and worker threads : https://github.com/vercel/next.js/issues/10933#issuecomment-598297975

## Prefetching in Preview Mode

- https://github.com/vercel/next.js/discussions/29058
- [https://github.com/vercel/next.js/discussions/24009](https://github.com/vercel/next.js/discussions/24009)

### **[This is now available in Next.js 12.1.0](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration#on-demand-revalidation-beta)**

- ~~https://github.com/vercel/next.js/discussions/11552~~
- ~~@leerob comment on the RFC: https://github.com/vercel/next.js/discussions/11552#discussioncomment-646963~~
- ~~https://github.com/vercel/next.js/discussions/22384 by @FSaldanha~~
- ~~https://github.com/vercel/next.js/pull/22699 by @adrianos10~~

**Comment by leerob** [comment]
Thanks for curating this list!

**Comment by charkour** [comment]
Thanks for this! On-demand revalidation for only pages whose content changed in the CMS would be a killer feature. Right now, it is frustrating to rebuild the entire site when we need to run a cache invalidation.
